kv/util/util.go

```diff
@@ -247,13 +360,22 @@ func logUpdateSuccess(logger xlog.Logger, k string, ver int, v interface{}) {
 	).Infof("value update success")
 }
 
+func logMalformedUpdate(logger xlog.Logger, k string, ver int, v interface{}, err error) {
+	getLogger(logger).WithFields(
+		xlog.NewLogField("key", k),
+		xlog.NewLogField("value", v),
+		xlog.NewLogField("version", ver),
+		xlog.NewLogField("error", err),
+	).Warnf("malformed value from kv store, use default value")
+}
+
 func logInvalidUpdate(logger xlog.Logger, k string, ver int, v interface{}, err error) {
 	getLogger(logger).WithFields(
 		xlog.NewLogField("key", k),
 		xlog.NewLogField("value", v),
 		xlog.NewLogField("version", ver),
 		xlog.NewLogField("error", err),
-	).Infof("invalid value from kv store, use default value")
+	).Warnf("invalid value from kv store, not applying update")
```

Comment:
The second part of the warning message should probably be consistent with `logMalformedUpdate`?

Reply:
We should discuss this I think. I don't feel that we should be resetting the value to the default value for malformed updates. That seems wrong to me since if the update is bad there really isn't any reason to apply the update at all in my opinion. And I feel even more strongly that we should not apply the default value for invalid updates. So if we are going to make the behavior consistent I'm in favor of changing the logic for malformed updates to not apply the default value.